storybookjs/storybook
# Next.js: Add next/font/local declarations support

Discussion: https://github.com/storybookjs/storybook/discussions/24979#discussioncomment-7666280




## What I did

Add support for next/font/local [declarations](https://nextjs.org/docs/pages/api-reference/components/font#declarations).

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Add `ascent-override: 500%` to `localRubikStorm` local font on the `code/frameworks/nextjs/template/stories_nextjs-default-js/Font.jsx` file.

```diff
export const localRubikStorm = localFont({
  src: '/fonts/RubikStorm-Regular.ttf',
  variable: '--font-rubik-storm',
+ declarations: [{ prop: 'ascent-override', value: '500%' }],
});
```
2. Start the `nexjs/default-js` sandbox:

```
yarn task --task dev --template nextjs/default-js --start-from=sandbox
```

3. Got to the `/story/stories-frameworks-nextjs-nextjs-default-js-font--with-class-name` story and select the text for the local font, you should be able to see how the ascent for the font was modified:
 
<img width="797" alt="image" src="https://github.com/storybookjs/storybook/assets/2121481/be51035f-5092-434d-8b5d-d5db31e47abf">



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
